MapboxNavigation/PollyVoiceController.swift

```diff
@@ -85,12 +91,14 @@ public class PollyVoiceController: RouteVoiceController {
             }
         }
         
-        guard spokenInstructionsForRoute.object(forKey: instruction as NSString) == nil else {
-            play(spokenInstructionsForRoute.object(forKey: instruction as NSString)! as Data)
+        guard spokenInstructionsForRoute.object(forKey: instruction.ssmlText as NSString) == nil else {
+            play(spokenInstructionsForRoute.object(forKey: instruction.ssmlText as NSString)! as Data)
             return
         }
         
-        speak(instruction, error: nil)
+        guard isEnabled, volume > 0, !NavigationSettings.shared.muted else { return }
+        
+        super.speak(instruction)
```

Comment:
This causes `RouteVoiceController.speak(_:)` to be called, so `PollyVoiceController.speak(_:)` never gets called.

An override should normally call the superclass’s implementation. So call `super.didPassSpokenInstructionPoint(_:)`. That will allow `RouteVoiceController.didPassSpokenInstructionPoint(_:)` to determine which instruction to pass into `PollyVoiceController.speak(_:)`.

If that back-and-forth concerns you, bring back `RouteVoiceController.shouldSpeak(for:)` and move this prefecthing logic to an override of that method, `PollyVoiceController.shouldSpeak(for:)`. (The override can call the super implementation after performing the prefetching.)

Reply:
Fixed in a84a28f2df09f12dbe2f8d6b8768b22b248d19d7.